### ISSUE TITLE
Timestamp equality broadcasts incorrectly

### ISSUE THREAD

**Issue by jcrist** [pydata/pandas#11034]
Always returns true for equality and inequality:

```python
In [168]: t = pd.Timestamp('2000-01-29 01:59:00')

In [169]: a = pd.Series([t])

In [170]: b = pd.Series([t])

In [171]: a
Out[171]:
0   2000-01-29 01:59:00
dtype: datetime64[ns]

In [172]: a == b
Out[172]:
0    True
dtype: bool

In [173]: a != b
Out[173]:
0    True
dtype: bool

In [174]: t == t
Out[174]: True

In [175]: t != t
Out[175]: False
```

**Comment by cpcloud** [comment]
I'm tracking this down.

**Comment by jreback** [comment]
hmm, this should work......2 big changes in 0.17....comparisions vs None and datetime with tz, both touched ``core/ops`` a lot

**Comment by jreback** [comment]
closed by #11036
